### ISSUE TITLE
Unable to install CuArrays.jl

### ISSUE THREAD

**Issue by mouryarahul** [JuliaGPU/CuArrays.jl#232]
I have installed CUDAnative.jl and it passes the Test correctly:

> Pkg.test("CUDAnative.jl")
Testing CUDAnative
 Resolving package versions...
    Status `/tmp/tmpMo6eJk/Manifest.toml`
  [79e6a3ab] Adapt v0.4.1
  [3895d2a7] CUDAapi v0.5.3
  [c5f51814] CUDAdrv v0.8.6
  [be33ccc6] CUDAnative v0.10.1
  [929cbde3] LLVM v1.0.0
  [2a0f44e3] Base64  [`@stdlib/Base64`]
  [ade2ca70] Dates  [`@stdlib/Dates`]
  [8ba89e20] Distributed  [`@stdlib/Distributed`]
  [b77e0a4c] InteractiveUtils  [`@stdlib/InteractiveUtils`]
  [76f85450] LibGit2  [`@stdlib/LibGit2`]
  [8f399da3] Libdl  [`@stdlib/Libdl`]
  [37e2e46d] LinearAlgebra  [`@stdlib/LinearAlgebra`]
  [56ddb016] Logging  [`@stdlib/Logging`]
  [d6f4376e] Markdown  [`@stdlib/Markdown`]
  [44cfe95a] Pkg  [`@stdlib/Pkg`]
  [de0858da] Printf  [`@stdlib/Printf`]
  [3fa0cd96] REPL  [`@stdlib/REPL`]
  [9a3f8284] Random  [`@stdlib/Random`]
  [ea8e919c] SHA  [`@stdlib/SHA`]
  [9e88b42a] Serialization  [`@stdlib/Serialization`]
  [6462fe0b] Sockets  [`@stdlib/Sockets`]
  [2f01184e] SparseArrays  [`@stdlib/SparseArrays`]
  [10745b16] Statistics  [`@stdlib/Statistics`]
  [8dfed614] Test  [`@stdlib/Test`]
  [cf7118a7] UUIDs  [`@stdlib/UUIDs`]
  [4ec0a83e] Unicode  [`@stdlib/Unicode`]
[ Info: Testing using device GeForce GTX 1070
Test Summary: | Pass  Total
CUDAnative    |  251    251
   Testing CUDAnative tests passed 

But, when I try to install CuArrays.jl, I get following errors:

> Pkg.add("CuArrays")
  Updating registry at `~/.julia/registries/General
  Updating git-repo `https://github.com/JuliaRegistries/General.git
 Resolving package versions...
ERROR: Unsatisfiable requirements detected for package Adapt [79e6a3ab]:
 Adapt [79e6a3ab] log:
 ├─possible versions are: [0.1.0, 0.2.0, 0.3.0-0.3.1, 0.4.0-0.4.1] or uninstalled
 ├─restricted by compatibility requirements with CUDAnative [be33ccc6] to versions: 0.4.0-0.4.1
 │ └─CUDAnative [be33ccc6] log:
 │   ├─possible versions are: [0.1.0-0.1.3, 0.2.0, 0.3.0-0.3.1, 0.4.0-0.4.2, 0.5.0-0.5.5, 0.6.0-0.6.3, 0.7.0, 0.8.0-0.8.10, 0.9.0-0.9.1, 0.10.0-0.10.1] or uninstalled
 │   └─restricted to versions 0.10.1 by an explicit requirement, leaving only versions 0.10.1
 └─restricted by compatibility requirements with CuArrays [3a865a2d] to versions: [0.1.0, 0.2.0, 0.3.0-0.3.1] — no versions left
   └─CuArrays [3a865a2d] log:
     ├─possible versions are: [0.1.0-0.1.1, 0.2.0-0.2.1, 0.3.0, 0.4.0, 0.5.0, 0.6.0-0.6.2, 0.7.0-0.7.3, 0.8.0-0.8.1] or uninstalled
     ├─restricted to versions * by an explicit requirement, leaving only versions [0.1.0-0.1.1, 0.2.0-0.2.1, 0.3.0, 0.4.0, 0.5.0, 0.6.0-0.6.2, 0.7.0-0.7.3, 0.8.0-0.8.1]
     └─restricted by compatibility requirements with CUDAnative [be33ccc6] to versions: 0.8.0-0.8.1 or uninstalled, leaving only versions: 0.8.0-0.8.1
       └─CUDAnative [be33ccc6] log: see above
Stacktrace:
 [1] #propagate_constraints!#61(::Bool, ::Function, ::Pkg.GraphType.Graph, ::Set{Int64}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Pkg/src/GraphType.jl:1005
 [2] propagate_constraints! at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Pkg/src/GraphType.jl:946 [inlined]
 [3] #simplify_graph!#121(::Bool, ::Function, ::Pkg.GraphType.Graph, ::Set{Int64}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Pkg/src/GraphType.jl:1460
 [4] simplify_graph! at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Pkg/src/GraphType.jl:1460 [inlined] (repeats 2 times)
 [5] macro expansion at ./logging.jl:317 [inlined]
 [6] resolve_versions!(::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}, ::Nothing) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Pkg/src/Operations.jl:353
 [7] resolve_versions! at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Pkg/src/Operations.jl:316 [inlined]
 [8] #add_or_develop#62(::Array{Base.UUID,1}, ::Symbol, ::Function, ::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Pkg/src/Operations.jl:1201
 [9] #add_or_develop at ./none:0 [inlined]
 [10] #add_or_develop#13(::Symbol, ::Bool, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Pkg/src/API.jl:64
 [11] #add_or_develop at ./none:0 [inlined]
 [12] #add_or_develop#12 at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Pkg/src/API.jl:29 [inlined]
 [13] #add_or_develop at ./none:0 [inlined]
 [14] #add_or_develop#11 at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Pkg/src/API.jl:28 [inlined]
 [15] #add_or_develop at ./none:0 [inlined]
 [16] #add_or_develop#10 at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Pkg/src/API.jl:27 [inlined]
 [17] #add_or_develop at ./none:0 [inlined]
 [18] #add#18 at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Pkg/src/API.jl:69 [inlined]
 [19] add(::String) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.0/Pkg/src/API.jl:69
 [20] top-level scope at none:0 

Any hint to resolve this problem? Thanks in advance. Btw the system info is:
> versioninfo()
Julia Version 1.0.2
Commit d789231e99 (2018-11-08 20:11 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i7-7820HK CPU @ 2.90GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.0 (ORCJIT, skylake)

**Comment by maleadt** [comment]
Due to manually installing CUDAnative first, it installed a version that isn't compatible with CuArrays. There is a combination of packages that just works though, so try removing CUDAnative, installing CuArrays, and adding CUDAnative afterwards. That should make Pkg stick to a compatible set of packages.

I would have expected Pkg to downgrade CUDAnative since you didn't require an explicit version. @MikeInnes, any progress on the Flux issues so that we can tag CuArrays?

**Comment by rana** [comment]
`remove CUDAnative` _and_ `remove CUDAdrv` then `add CuArrays` also worked.

**Comment by ZhenhongDu** [comment]
How to solve this problem?I did not add CUDAnative package
`ERROR: Unsatisfiable requirements detected for package CuArrays [3a865a2d]:

```

> ERROR: Unsatisfiable requirements detected for package CuArrays [3a865a2d]:
>  CuArrays [3a865a2d] log:
>  ├─possible versions are: [0.2.1, 0.3.0, 0.4.0, 0.5.0, 0.6.0-0.6.2, 0.7.0-0.7.3, 0.8.0-0.8.1, 0.9.0-0.9.1, 1.0.0-1.0.2, 1.1.0, 1.2.0-1.2.1, 1.3.0, 1.4.0-1.4.7, 1.5.0, 1.6.0, 1.7.0-1.7.3, 2.0.0-2.0.1, 2.1.0, 2.2.0-2.2.2] or uninstalled
>  ├─restricted to versions * by an explicit requirement, leaving only versions [0.2.1, 0.3.0, 0.4.0, 0.5.0, 0.6.0-0.6.2, 0.7.0-0.7.3, 0.8.0-0.8.1, 0.9.0-0.9.1, 1.0.0-1.0.2, 1.1.0, 1.2.0-1.2.1, 1.3.0, 1.4.0-1.4.7, 1.5.0, 1.6.0, 1.7.0-1.7.3, 2.0.0-2.0.1, 2.1.0, 2.2.0-2.2.2]
>  ├─restricted by compatibility requirements with Adapt [79e6a3ab] to versions: [0.2.1, 0.3.0] or uninstalled, leaving only versions: [0.2.1, 0.3.0]
>  │ └─Adapt [79e6a3ab] log:
>  │   ├─possible versions are: [0.3.0-0.3.1, 0.4.0-0.4.2, 1.0.0-1.0.1, 1.1.0, 2.0.0-2.0.2] or uninstalled
>  │   └─restricted by compatibility requirements with CUDA [052768ef] to versions: 2.0.0-2.0.2
>  │     └─CUDA [052768ef] log:
>  │       ├─possible versions are: [0.1.0, 1.0.0-1.0.2, 1.1.0, 1.2.0-1.2.1] or uninstalled
>  │       └─restricted to versions 1.2.0-1 by ThreeDeconv [8923caa7], leaving only versions 1.2.0-1.2.1
>  │         └─ThreeDeconv [8923caa7] log:
>  │           ├─possible versions are: 1.3.0 or uninstalled
>  │           └─ThreeDeconv [8923caa7] is fixed to version 1.3.0
>  └─restricted by compatibility requirements with CUDAdrv [c5f51814] to versions: [0.4.0, 0.5.0, 0.6.0-0.6.2, 0.7.0-0.7.3, 0.8.0-0.8.1] or uninstalled — no versions left
>    └─CUDAdrv [c5f51814] log:
>      ├─possible versions are: [0.8.0-0.8.6, 0.9.0, 1.0.0-1.0.1, 2.0.0, 3.0.0-3.0.1, 3.1.0, 4.0.0-4.0.4, 5.0.0-5.0.1, 5.1.0, 6.0.0-6.0.1, 6.1.0, 6.2.0-6.2.3, 6.3.0] or uninstalled
>      ├─restricted by compatibility requirements with CuArrays [3a865a2d] to versions: [0.8.0-0.8.6, 1.0.1, 2.0.0, 3.0.0-3.0.1, 3.1.0, 4.0.1-4.0.4, 5.0.0-5.0.1, 5.1.0, 6.0.0-6.0.1, 6.1.0, 6.2.0-6.2.3, 6.3.0]
>      │ └─CuArrays [3a865a2d] log: see above
>      ├─restricted by compatibility requirements with Compat [34da2185] to versions: [0.8.1-0.8.6, 0.9.0, 1.0.0-1.0.1, 2.0.0, 3.0.0-3.0.1, 3.1.0, 4.0.0-4.0.4, 5.0.0-5.0.1, 5.1.0, 6.0.0-6.0.1, 6.1.0, 6.2.0-6.2.3, 6.3.0] or uninstalled, leaving only versions: [0.8.1-0.8.6, 1.0.1, 2.0.0, 3.0.0-3.0.1, 3.1.0, 4.0.1-4.0.4, 5.0.0-5.0.1, 5.1.0, 6.0.0-6.0.1, 6.1.0, 6.2.0-6.2.3, 6.3.0]
>      │ └─Compat [34da2185] log:
>      │   ├─possible versions are: [1.0.0-1.0.1, 1.1.0, 1.2.0, 1.3.0, 1.4.0, 1.5.0-1.5.1, 2.0.0, 2.1.0, 2.2.0, 3.0.0, 3.1.0, 3.2.0, 3.3.0-3.3.1, 3.4.0, 3.5.0, 3.6.0, 3.7.0, 3.8.0, 3.9.0-3.9.1, 3.10.0, 3.11.0, 3.12.0, 3.13.0] or uninstalled
>      │   ├─restricted to versions 0.22.0-* by Lint [d4b2101a], leaving only versions [1.0.0-1.0.1, 1.1.0, 1.2.0, 1.3.0, 1.4.0, 1.5.0-1.5.1, 2.0.0, 2.1.0, 2.2.0, 3.0.0, 3.1.0, 3.2.0, 3.3.0-3.3.1, 3.4.0, 3.5.0, 3.6.0, 3.7.0, 3.8.0, 3.9.0-3.9.1, 3.10.0, 3.11.0, 3.12.0, 3.13.0]
>      │   │ └─Lint [d4b2101a] log:
>      │   │   ├─possible versions are: 0.0.0 or uninstalled
>      │   │   └─Lint [d4b2101a] is fixed to version 0.0.0
>      │   └─restricted by compatibility requirements with Optim [429524aa] to versions: [3.2.0, 3.3.0-3.3.1, 3.4.0, 3.5.0, 3.6.0, 3.7.0, 3.8.0, 3.9.0-3.9.1, 3.10.0, 3.11.0, 3.12.0, 3.13.0]
>      │     └─Optim [429524aa] log:
>      │       ├─possible versions are: [0.15.3, 0.16.0, 0.17.0-0.17.2, 0.18.0-0.18.1, 0.19.0-0.19.7, 0.20.0-0.20.6, 0.21.0, 0.22.0] or uninstalled
>      │       └─restricted to versions 0.22 by ThreeDeconv [8923caa7], leaving only versions 0.22.0
>      │         └─ThreeDeconv [8923caa7] log: see above
>      └─restricted by compatibility requirements with CUDAnative [be33ccc6] to versions: 0.8.0-0.8.6, leaving only versions: 0.8.1-0.8.6
>        └─CUDAnative [be33ccc6] log:
>          ├─possible versions are: [0.7.0, 0.8.0-0.8.10, 0.9.0-0.9.1, 0.10.0-0.10.1, 1.0.0-1.0.1, 2.0.0-2.0.1, 2.1.0-2.1.3, 2.2.0-2.2.1, 2.3.0-2.3.1, 2.4.0, 2.5.0-2.5.5, 2.6.0, 2.7.0, 2.8.0-2.8.1, 2.9.0-2.9.1, 2.10.0-2.10.2, 3.0.0-3.0.4, 3.1.0, 3.2.0] or uninstalled
>          ├─restricted by compatibility requirements with CuArrays [3a865a2d] to versions: [0.7.0, 0.8.0-0.8.10, 0.9.0-0.9.1, 0.10.0-0.10.1, 1.0.0-1.0.1, 2.0.0-2.0.1, 2.1.0-2.1.3, 2.2.0-2.2.1, 2.3.0-2.3.1, 2.4.0, 2.5.0-2.5.5, 2.6.0, 2.7.0, 2.8.0-2.8.1, 2.9.0-2.9.1, 2.10.0-2.10.2, 3.0.0-3.0.4, 3.1.0, 3.2.0]
>          │ └─CuArrays [3a865a2d] log: see above
>          ├─restricted by compatibility requirements with GPUCompiler [61eb1bfa] to versions: [0.7.0, 0.8.0-0.8.10, 0.9.0-0.9.1, 0.10.0-0.10.1, 1.0.0-1.0.1, 2.0.0-2.0.1, 2.1.0-2.1.3, 2.2.0-2.2.1, 2.3.0-2.3.1, 2.4.0, 2.5.0-2.5.5, 2.6.0, 2.7.0, 2.8.0-2.8.1, 2.9.0-2.9.1, 2.10.0-2.10.2, 3.0.0-3.0.4] or uninstalled, leaving only versions: [0.7.0, 0.8.0-0.8.10, 0.9.0-0.9.1, 0.10.0-0.10.1, 1.0.0-1.0.1, 2.0.0-2.0.1, 2.1.0-2.1.3, 2.2.0-2.2.1, 2.3.0-2.3.1, 2.4.0, 2.5.0-2.5.5, 2.6.0, 2.7.0, 2.8.0-2.8.1, 2.9.0-2.9.1, 2.10.0-2.10.2, 3.0.0-3.0.4]
>          │ └─GPUCompiler [61eb1bfa] log:
>          │   ├─possible versions are: [0.1.0, 0.2.0, 0.3.0, 0.4.0-0.4.1, 0.5.0-0.5.5] or uninstalled
>          │   └─restricted by compatibility requirements with CUDA [052768ef] to versions: [0.4.0-0.4.1, 0.5.0-0.5.5]
>          │     └─CUDA [052768ef] log: see above
>          └─restricted by compatibility requirements with Adapt [79e6a3ab] to versions: [0.7.0, 0.8.0-0.8.10, 0.9.0-0.9.1] or uninstalled, leaving only versions: [0.7.0, 0.8.0-0.8.10, 0.9.0-0.9.1]
>            └─Adapt [79e6a3ab] log: see above

```

**Comment by maleadt** [comment]
Please don't revive old issues like that.

You are (probably indirectly) depending on both CUDA.jl and CuArrays.jl; this is not supported.
